### ISSUE TITLE
core-ajax-dart throws exception when used with no params attribute

### ISSUE THREAD

**Issue by DartBot** [dart-lang/polymer-dart#235]
_Originally opened as dart-lang/sdk#19967_

*This issue was originally filed by seandia...&#064;google.com*
_____
**What steps will reproduce the problem?**

Run the following in Dartium:

&lt;!doctype html&gt;
&lt;html&gt;
&nbsp;&nbsp;&lt;head&gt;
&nbsp;&nbsp;&nbsp;&nbsp;&lt;title&gt;core-ajax-dart param bug&lt;/title&gt;

&nbsp;&nbsp;&nbsp;&nbsp;&lt;link rel=&quot;import&quot; href=&quot;packages/polymer/polymer.html&quot;&gt;
&nbsp;&nbsp;&nbsp;&nbsp;&lt;link rel=&quot;import&quot; href=&quot;packages/core_elements/core_ajax_dart.html&quot;&gt;
&nbsp;&nbsp;&lt;/head&gt;

&nbsp;&nbsp;&lt;body&gt;
&nbsp;&nbsp;&nbsp;&nbsp;&lt;!-- This fails unless a non-empty params attribute is passed in (e.g. params=&quot;{}&quot;) --&gt;
&nbsp;&nbsp;&nbsp;&nbsp;&lt;core-ajax-dart handleAs=&quot;json&quot; url=&quot;http://date.jsontest.com/&quot;&gt;&lt;/core-ajax-dart&gt;

&nbsp;&nbsp;&nbsp;&nbsp;&lt;script type=&quot;application/dart&quot;&gt;
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;import 'dart:html';
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;import 'package:polymer/polymer.dart';

&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;main() {
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;initPolymer().run(() {
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;try {
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;querySelector('core-ajax-dart').go();
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;} catch(x, st) {
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;print(st);
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;}
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;});
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;}
&nbsp;&nbsp;&nbsp;&nbsp;&lt;/script&gt;
&nbsp;&nbsp;&lt;/body&gt;
&lt;/html&gt;

**What is the expected output? What do you see instead?**

I expect to see a request to date.jsontest.com in the Network tab of the dev tools panel. Instead, I see the following stack trace in the console:

CoreAjax.created (:1)
#­0      \_JsonParser.fail (dart:convert-patch/convert_patch.dart:580)
#­1      \_JsonParser.parse (dart:convert-patch/convert_patch.dart:336)
#­2      \_parseJson (dart:convert-patch/convert_patch.dart:16)
#­3      JsonDecoder.convert (dart:convert/json.dart:321)
#­4      JsonCodec.decode (dart:convert/json.dart:124)
#­5      CoreAjax.go (package:core_elements/core_ajax_dart.dart:270:29)
#­6      main.&lt;anonymous closure&gt; (http://127.0.0.1:8080/index.html.0.dart:9:47)
#­7      \_rootRun (dart:async/zone.dart:719)
#­8      \_RootZone.run (dart:async/zone.dart:862)
#­9      main (http://127.0.0.1:8080/index.html.0.dart:7:26)
#­10     main (http://127.0.0.1:8080/index.html_bootstrap.dart:88:10)

Adding a non-empty params attribute (e.g. params=&quot;{}&quot;) to the core-ajax-dart element fixes the problem and causes the request to succeed.

**Please include the following information:**
- Dart SDK version: 1.5.3
- polymer package version: 0.11.0+5
- Operating system: Mac OS X 10.9.3
- Browser: Dartium (bundled with SDK)

**Comment by DartBot** [comment]
<a href="https://github.com/sigmundch"><img src="https://avatars.githubusercontent.com/u/2049220?v=3" align="left" width="48" height="48"hspace="10"></img></a> **Comment by [sigmundch](https://github.com/sigmundch)**

----

Thanks for the bug report. This seems like a bug in the core-ajax-dart element itself, so I'm moving the bug to where that code lives: https://github.com/dart-lang/core-elements/issues/84
______
*Added [MovedToGithub](../labels/MovedToGithub) label.*

**Comment by DartBot** [comment]
<a href="https://github.com/jakemac53"><img src="https://avatars.githubusercontent.com/u/984921?v=3" align="left" width="48" height="48"hspace="10"></img></a> **Comment by [jakemac53](https://github.com/jakemac53)**

----

fixed in https://github.com/dart-lang/core-elements/pull/91, released as 0.1.1+1
